# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/linux26/12-conf-hacks.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./scripts/kconfig/conf.c.orig	2003-01-13 17:11:05.000000000 +0100
+++ ./scripts/kconfig/conf.c	2003-01-13 23:20:21.000000000 +0100
@@ -20,6 +20,7 @@
 	ask_all,
 	ask_new,
 	ask_silent,
+	ask_no2mod,
 	set_default,
 	set_yes,
 	set_mod,
@@ -102,6 +103,15 @@
 	line[1] = 0;
 
 	switch (input_mode) {
+	case ask_no2mod:
+		if (type == S_TRISTATE &&
+		    sym_get_tristate_value(sym) == no) {
+			fprintf(stderr, "Setting %s to 'm'.\n", sym->name);
+			line[0] = 'm';
+			line[1] = '\n';
+			line[2] = 0;
+			break;
+		}
 	case ask_new:
 	case ask_silent:
 		if (sym_has_value(sym)) {
@@ -354,6 +354,7 @@
 				break;
 			}
 			check_stdin();
+		case ask_no2mod:
 		case ask_all:
 			fflush(stdout);
 			fgets(line, 128, stdin);
@@ -506,6 +517,9 @@
 		case 'm':
 			input_mode = set_mod;
 			break;
+		case 'M':
+			input_mode = ask_no2mod;
+			break;
 		case 'y':
 			input_mode = set_yes;
 			break;
@@ -545,7 +559,8 @@
 		}
 	case ask_all:
 	case ask_new:
+	case ask_no2mod:
 		conf_read(NULL);
 		break;
 	default:


--- ./Makefile.orig	2003-05-05 01:53:34.000000000 +0200
+++ ./Makefile	2003-06-07 17:41:56.000000000 +0200
@@ -204,7 +204,7 @@
 
 noconfig_targets := xconfig gconfig menuconfig config oldconfig randconfig \
 		    defconfig allyesconfig allnoconfig allmodconfig \
-		    clean mrproper distclean rpm \
+		    no2modconfig clean mrproper distclean rpm \
 		    help tags TAGS cscope sgmldocs psdocs pdfdocs htmldocs \
 		    checkconfig checkhelp checkincludes
 
@@ -636,6 +636,9 @@
 allmodconfig: scripts/kconfig/conf
 	./scripts/kconfig/conf -m arch/$(ARCH)/Kconfig
 
+no2modconfig: scripts/kconfig/conf
+	./scripts/kconfig/conf -M arch/$(ARCH)/Kconfig
+
 defconfig: scripts/kconfig/conf
 	./scripts/kconfig/conf -d arch/$(ARCH)/Kconfig
 
